example/src/main/java/com/vimeo/example/MainActivity.kt

```diff
@@ -0,0 +1,129 @@
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package com.vimeo.example
+
+import android.content.Intent
+import android.net.Uri
+import android.os.Bundle
+import android.widget.Toast
+import androidx.appcompat.app.AppCompatActivity
+import com.vimeo.networking2.*
+import com.vimeo.networking2.config.VimeoApiConfiguration
+import kotlinx.android.synthetic.main.activity_main.*
+
+/**
+ * The sample activity in which we create an [Authenticator] instance to authenticate with the server.
+ */
+class MainActivity : AppCompatActivity() {
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_main)
+
+        // Construct a configuration instance.
+        val configuration = VimeoApiConfiguration.Builder(CLIENT_ID, CLIENT_SECRET, listOf(ScopeType.PUBLIC))
+            .withCodeGrantRedirectUrl("https://example.com")
+            .build()
+
+        // Initialize the authenticator instance with the configuration and the api client.
+        val authenticator = Authenticator(configuration)
+        val apiClient = VimeoApiClient(configuration, authenticator)
+
+        // If we are handling a code grant redirect, this URI will be present, otherwise it's a normal app start.
+        val redirectUri = intent?.data
+
+        if (redirectUri != null) {
+            // Log in using the code grant URI provided to us by the redirect.
+            authenticator.authenticateWithCodeGrant(redirectUri.toString(), vimeoCallback(
+                onSuccess = {
+                    toast("Successfully logged in.")
+                    loginStatus.text = authenticator.currentAccount.asLoginStatus()
+                },
+                onError = { toast("Unable to log in.") }
+            ))
+        }
+
+        loginStatus.text = authenticator.currentAccount.asLoginStatus()
+
+        clientCredentials.setOnClickListener {
+            // Obtain logged out credentials which can be used to make logged out requests.
+            authenticator.clientCredentials(vimeoCallback(
+                onSuccess = {
+                    toast("Obtained client credentials.")
+                    loginStatus.text = authenticator.currentAccount.asLoginStatus()
+                },
+                onError = { toast("Couldn't obtain client credentials") }
+            ))
+        }
+
+        login.setOnClickListener {
```

Comment:
I'm getting an error when I try this. 
![Screenshot_1597088056](https://user-images.githubusercontent.com/3968678/89823348-54b77600-db1f-11ea-8ca9-f95bba5ee0b4.png)

Reply:
Not sure what changes you made to the example before running it, but because the docs aren't clear - you need to create an API app here: https://developer.vimeo.com/apps/new and then register the redirect URI (in the example it's set to `https://example.com`) on the API page - navigate to https://developer.vimeo.com/apps and then choose your app and at the bottom click "add URL." Once this is done replace the client id and client secret placeholders in the example.

Reply:
I was setting up the code grant URL wrong, updated docs helped me get it working properly. 👍